pallets/pips/src/lib.rs

```diff
@@ -867,7 +897,7 @@ decl_module! {
         /// * `ProposalOnCoolOffPeriod` if non-owner is voting and PIP is cooling off.
         /// * `IncorrectProposalState` if PIP isn't pending.
         /// * `InsufficientDeposit` if `origin` cannot reserve `deposit - old_deposit`.
-        #[weight = 1_000_000_000]
+        #[weight = <T as Trait>::WeightInfo::vote(1_000, 10)]
```

Comment:
None of these numbers have a big practical impact. Let's remove them from here and use constants in the benchmarks.

Reply:
@maxsam4 Constant quantities apparently lead to heavy weights. Check out [these](https://github.com/PolymathNetwork/Polymesh/pull/722/files#diff-8e6cde2ec6f38a86bd7e544e00dd3f1e894e166db6779fee2c0d889cb1dca962R95-R114). Integration tests start to fail because of those weights. I'll lower down the constants but we should aim at bringing these parameters to the weight function in the near future.

Reply:
`snapshot` and `enact_snapshot_results` are expected to be dependent on these variables and that's why we are refactoring them (in a different PR). It does not make sense for `execute_scheduled_pip` and `expire_scheduled_pip` to be this dependent on other proposals though. Can you please see why that's happening and is there something we can do to avoid the dependency?

Reply:
I am going to guess that it is because of `<ProposalVotes<T>>::remove_prefix(id);`. This raises a question: Given that this operation frees up disk space in exchange for computation time, should we make it "free"? I think we should because we have plenty of headroom in computation time but not much in disk growth in Polymesh. 

I'd suggest that we add these storage keys to the whitelist and don't count them in the weight function. WDYT?
cc @Centril @adamdossa.

Reply:
Indeed, the number of votes is a major factor. Let me try with the whitelisted storage keys for `ProposalVotes` and `Deposits` - these are the only ones removed with `remove_prefix` AFAIK.

Reply:
@maxsam4 I'm convinced now that, in the current benchmark, the line with `ProposalVotes` is not executed because `PruneHistoricalPips` is `false`. However, as I mentioned, the other `remove_prefix` call is made, and that is to `Deposits`.

Action items for me:
1. set `PruneHistoricalPips` to `true` to increase complexity.
2. whitelist these two storage keys.

Reply:
It appears that the bench framework doesn't allow whitelisting of storage prefixes. So we have to whitelist 40000 deposit keys and 40000 proposal vote keys.